python/GafferImageTest/ResizeTest.py

```diff
@@ -0,0 +1,136 @@
+		c2[GafferImage.Format.defaultFormatContextName] = GafferImage.FormatData( f2 )
+
+		with c1 :
+			self.assertEqual( r["out"]["format"].getValue(), f1 )
+
+		with c2 :
+			self.assertEqual( r["out"]["format"].getValue(), f2 )
+
+	def testChannelDataPassThrough( self ) :
+
+		# Resize to the same size as the input image.
+		c = GafferImage.Constant()
+		c["format"].setValue( GafferImage.Format( IECore.Box2i( IECore.V2i( 0 ), IECore.V2i( 512 ) ), 1 ) )
+		c["color"].setValue( IECore.Color4f( 0.25, 0.5, 0.75, 1 ) )
+
+		r = GafferImage.Resize()
+		r["in"].setInput( c["out"] )
+		r["format"].setValue( GafferImage.Format( IECore.Box2i( IECore.V2i( 0 ), IECore.V2i( 512 ) ), 1 ) )
+
+		# Assert that the pixel data is passed clean through.
+		for channel in [ "R", "G", "B", "A" ] :
+			self.assertEqual(
+				c["out"].channelDataHash( channel, IECore.V2i( 0 ) ),
+				r["out"].channelDataHash( channel, IECore.V2i( 0 ) ),
+			)
+			self.assertTrue(
+				c["out"].channelData( channel, IECore.V2i( 0 ), _copy = False ).isSame(
+					r["out"].channelData( channel, IECore.V2i( 0 ), _copy = False )
+				)
+			)
+
+	def testFit( self ) :
+
+		c = GafferImage.Constant()
+		c["format"].setValue( GafferImage.Format( IECore.Box2i( IECore.V2i( 0 ), IECore.V2i( 256, 128 ) ), 1 ) )
+		c["color"].setValue( IECore.Color4f( 0.25, 0.5, 0.75, 1 ) )
+
+		r = GafferImage.Resize()
+		r["in"].setInput( c["out"] )
+		r["format"].setValue( GafferImage.Format( IECore.Box2i( IECore.V2i( 0 ), IECore.V2i( 1024, 256 ) ), 1 ) )
+
+		self.assertEqual( r["fit"].getValue(), r.Fit.Horizontal )
+
+		horizontalDataWindow = r["out"]["dataWindow"].getValue()
+		displayWindow = r["format"].getValue().getDisplayWindow()
+
+		self.assertEqual( horizontalDataWindow.min.x, displayWindow.min.x )
+		self.assertEqual( horizontalDataWindow.max.x, displayWindow.max.x )
+		self.assertTrue( horizontalDataWindow.min.y < displayWindow.min.y )
+		self.assertTrue( horizontalDataWindow.max.y > displayWindow.max.y )
+
+		r["fit"].setValue( r.Fit.Vertical )
+		verticalDataWindow = r["out"]["dataWindow"].getValue()
+
+		self.assertTrue( verticalDataWindow.min.x > displayWindow.min.x )
+		self.assertTrue( verticalDataWindow.max.x < displayWindow.max.x )
+		self.assertEqual( verticalDataWindow.min.y, displayWindow.min.y )
+		self.assertEqual( verticalDataWindow.max.y, displayWindow.max.y )
+
+		r["fit"].setValue( r.Fit.Best )
+		self.assertEqual( r["out"]["dataWindow"].getValue(), verticalDataWindow )
+
+		r["fit"].setValue( r.Fit.Fill )
+		self.assertEqual( r["out"]["dataWindow"].getValue(), horizontalDataWindow )
+
+		r["fit"].setValue( r.Fit.Distort )
+		self.assertEqual( r["out"]["dataWindow"].getValue(), displayWindow )
+
+	def testFilterAffectsChannelData( self ) :
+
+		 r = GafferImage.Resize()
+		 cs = GafferTest.CapturingSlot( r.plugDirtiedSignal() )
+		 r["filter"].setValue( "gaussian" )
+
+		 self.assertTrue( r["out"]["channelData"] in set( c[0] for c in cs ) )
+
+if __name__ == "__main__":
```

Comment:
There doesn't seem to be a test for what happens when the data window and the display window don't line up. Might be worth checking that case too